WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardReaderManualsViewModelTests.swift

```diff
@@ -0,0 +1,107 @@
+import XCTest
+import Yosemite
+@testable import WooCommerce
+
+class CardReaderManualsViewModelTests: XCTestCase {
+
+    private var storageManager: MockStorageManager!
+    private var stores: MockStoresManager!
+    private let sampleSiteID: Int64 = 1234
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        storageManager = MockStorageManager()
+        stores = MockStoresManager(sessionManager: .makeForTesting(authenticated: true))
+        stores.sessionManager.setStoreId(sampleSiteID)
+        ServiceLocator.setSelectedSiteSettings(SelectedSiteSettings(stores: stores, storageManager: storageManager))
+    }
+
+    override func tearDownWithError() throws {
+        ServiceLocator.setSelectedSiteSettings(SelectedSiteSettings())
+        storageManager.reset()
+        storageManager = nil
+        stores = nil
+        try super.tearDownWithError()
+    }
+
+    func test_viewModel_not_nil() {
+        // Given
+        let viewModel = CardReaderManualsViewModel()
+
+        // Then
+        XCTAssertNotNil(viewModel)
+    }
+
+    func test_viewModel_when_init_then_has_default_manuals() {
+         // Given
+         let viewModel = CardReaderManualsViewModel()
+
+         // When
+         let expectedManuals = viewModel.manuals
+
+         // Then
+         XCTAssertEqual(viewModel.manuals, expectedManuals)
```

Comment:
```suggestion
         // Given
        let viewModel = CardReaderManualsViewModel()
        let expectedManuals = [CardReaderType.chipper.manual, CardReaderType.stripeM2.manual]

        // When
        let actualManuals = viewModel.manuals

        // Then
        assertEqual(expectedManuals, actualManuals)
```

Slight changes here:
1. This defines the expectedManuals in a way that you can read what they are, and avoids later comparing that `viewModel.manuals == viewModel.manuals`, which will always be true.
2. We put `expectedManuals` in the Given section: it's an axiom, not an action
3. We're testing `viewModel.manuals`, so we put that in the When section
4. We can use the `assertEqual` helper, rather than `XCTAssertEqual`, which will tell us what the differences are between the two when the test fails.

Reply:
When I run this test, I get the following:
<img width="960" alt="CleanShot 2022-07-11 at 15 48 12@2x" src="https://user-images.githubusercontent.com/2472348/178292150-16683c5e-c0cd-4125-9859-fa3fddad73ff.png">

> test failure: CardReaderManualsViewModelTests.test_viewModel_when_init_then_has_default_manuals() failed: assertEqual failed: Found difference for 
> Different count:
> |	Received: (0) []
> |	Expected: (2) [ ... ]

This will have passed for you because the `viewModel.manuals` returned 0 elements in both cases.

```
po configurationLoader.configuration
▿ CardPresentPaymentsConfiguration
  - countryCode : ""
  - paymentMethods : 0 elements
  - currencies : 0 elements
  - paymentGateways : 0 elements
  - supportedReaders : 0 elements
```

The fix is to set a configuration, which you can pass in to `CardReaderManualsViewModel` in the init method, and allow it to use the default configuration at runtime, which we know works from our manual testing. e.g.:

```
// Given
let viewModel = CardReaderManualsViewModel(cardPresentConfiguration: CardPresentPaymentsConfiguration(country: "US"))
```

Reply:
Thanks! This is very helpful!

> avoids later comparing that viewModel.manuals == viewModel.manuals, which will always be true.

Oops, this one went unnoticed 😅